pkg/products/rhsso/reconciler.go

```diff
@@ -683,8 +682,44 @@ func (r *Reconciler) setupGithubIDP(ctx context.Context, kcr *keycloak.KeycloakR
 			},
 		})
 	}
-	// We need to revisit how the github idp gets created/updated
-	// client ID and secret can get outdated we need to ensure they are synced with the value secret in the github-oauth-secret
+
+	githubClientID := string(githubCreds.Data["clientId"])
+	githubClientSecret := string(githubCreds.Data["secret"])
+
+	// check if GH credentials have been set up
+	githubMockCred := "dummy"
+	if githubClientID == githubMockCred || githubClientSecret == githubMockCred {
+		return nil
+	}
+
+	logrus.Infof("Syncing github identity provider to the keycloak realm")
+
+	// Get an authenticated keycloak api client for the instance
+	keycloakFactory := common.LocalConfigKeycloakFactory{}
+	authenticated, err := keycloakFactory.AuthenticatedClient(*kc)
+	if err != nil {
+		return fmt.Errorf("Unable to authenticate to the Keycloak API: %s", err)
+	}
+
+	identityProvider, err := authenticated.GetIdentityProvider(githubIdpAlias, kcr.Spec.Realm.DisplayName)
+	if err != nil {
+		return fmt.Errorf("Unable to get Identity Provider from Keycloak API: %s", err)
+	}
+
+	// checking if GH credentials have changed
+	if githubClientID == identityProvider.Config["clientId"] && githubClientSecret == identityProvider.Config["secret"] {
```

Comment:
`githubClientSecret == identityProvider.Config["secret"]` will never be true because Keycloak API always returns `********` for sensitive values, such as this secret. Screenshot with evidence below :)
So I think this condition can be simply removed.

![Screenshot 2020-01-30 at 14 27 06](https://user-images.githubusercontent.com/1605799/73453800-115c9d80-436d-11ea-8445-c9f9c24ff9b7.png)

Reply:
I didn't know that, well spotted. Thanks